mtcpclient/groupofconnections.go

```diff
@@ -29,23 +48,76 @@ func (gc GroupOfConnections) String() string {
 		nTotal, nDialing, nEstablished, nClosed, nError, nNotInitiated)
 }
 
-func (gc GroupOfConnections) isIn(status tcpclient.ConnectionStatus) bool {
-	for _, item := range gc {
-		if item.GetConnectionStatus() == status {
-			return true
+func (gc GroupOfConnections) containsAConnectionWithStatus(status string) bool {
+	for _, connection := range gc.connections {
+		switch status {
+		case "pending":
+			if connection.PendingToProcess() {
```

Comment:
why this level of indirection (if... -> return true, while the if condition already returns true)?

Reply:
I used this to break the look once a condition is matched, depending on each case